### PR TITLE
feat: Add lpAprs json file and github action to update it

### PR DESCRIPTION
**Pull request by Chef-Cheems**
This PR adds `src/config/constants/lpAPRs.json` that stores LP rewards APR estimation.
GitHub Action is added to update this file every day at 23:15 UTC (GitHub suggests not to set time at exact hour to prevent delays due to load. Also scheduled runs are usually delayed in any case, but should run somewhere between 23:15-00:00)

GitHub action will create a PR for updated values that we can review for potential errors (e.g. huge APRs, zeros, etc.)

PR also includes initial fetch for lpAprs.json

### CODE REVIEW

File: scripts/updateLPsAPR/index.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  return `pairs(
+      first: 30,
+      where: { id_in: ${addressesString} }
+      ${blockString}
+    ) {
+      id
+      volumeUSD
+      reserveUSD
+    }`
+}
```
Comment (hachiojidev) [inline scripts/updateLPsAPR/index.ts:31]: This is kind of dirty. I think we should use `graphql-request` with `gql`.

File: scripts/updateLPsAPR/index.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+const FARMS_BULK = (blockWeekAgo: number, farms: string[]) => {
```
Comment (hachiojidev) [inline scripts/updateLPsAPR/index.ts:33]: Can we rename this function to something more descriptive?

File: scripts/updateLPsAPR/index.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+interface AprMap {
+  [key: string]: BigNumber
+}
+
+const getWeekAgoTimestamp = () => {
+  const weekAgoMs = new Date().getTime() - 7 * 24 * 60 * 60 * 1000
+  const weekAgoUnix = new Date(weekAgoMs).getTime() / 1000
+  return Math.floor(weekAgoUnix).toString()
```
Comment (hachiojidev) [inline scripts/updateLPsAPR/index.ts:66]: This is totally fine but consider using `date-fns`, we have it available.

File: scripts/updateLPsAPR/index.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+const LP_HOLDERS_FEE = 0.0017
+const WEEKS_IN_A_YEAR = 52.1429
+
+const get7dBlock = async () => {
```
Comment (hachiojidev) [inline scripts/updateLPsAPR/index.ts:72]: I think we should avoid methods like this. Something more generic would be easier to maintain.

File: scripts/updateLPsAPR/index.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+interface SingleFarmResponse {
+  id: string
+  reserveUSD: string
+  volumeUSD: string
+}
+
+interface FarmsResponse {
+  now: SingleFarmResponse[]
```
Comment (hachiojidev) [inline scripts/updateLPsAPR/index.ts:55]: now?

File: scripts/updateLPsAPR/index.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    }, {})
+    return aprs
+  } catch (error) {
+    throw new Error(`Failed to fetch LP APR data: ${error}`)
+  }
+}
+
+const fetchAndUpdateLPsAPR = async () => {
+  const lowerCaseAddresses = farmsConfig
+    .filter((farm) => farm.pid > 250)
```
Comment (hachiojidev) [inline scripts/updateLPsAPR/index.ts:109]: At the very least this needs a comment explaining what we are filtering here.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: scripts/updateLPsAPR/index.ts (hunk 5 above)
Comment (Chef-Cheems) [reply scripts/updateLPsAPR/index.ts:55]: now as in "at latest available block right now"

File: scripts/updateLPsAPR/index.ts (hunk 1 above)
Comment (Chef-Cheems) [reply scripts/updateLPsAPR/index.ts:31]: This is mostly to keep stuff DRY, but I guess in this case as we don't request much I can just verbosely put everything into query below without this helper function 👍

File: scripts/updateLPsAPR/index.ts (hunk 5 above)
Comment (Chef-Cheems) [reply scripts/updateLPsAPR/index.ts]: renamed to `farmsAtLatestBlock`

File: scripts/updateLPsAPR/index.ts (hunk 4 above)
Comment (Chef-Cheems) [reply scripts/updateLPsAPR/index.ts]: refactored to `getBlockAtTimestamp = async (timestamp: string) => {...`

File: scripts/updateLPsAPR/index.ts (hunk 3 above)
Comment (Chef-Cheems) [reply scripts/updateLPsAPR/index.ts]: For some reason I forgot we had it as dependency to be honest 😅

File: scripts/updateLPsAPR/index.ts (hunk 2 above)
Comment (Chef-Cheems) [reply scripts/updateLPsAPR/index.ts]: Removed it completely query is small enough to just put it directly as argument to graphql-request

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**